src/components/common/PropTypes.ts

```diff
@@ -0,0 +1,6 @@
+export interface DynamicListProps<TModel> {
+  addText: string | ((listItems: Array<TModel>) => string);
+  modelFactory: () => TModel;
```

Comment:
emptyModelFactory? defaultModelFactory? createEmptyModel?

Reply:
I am not sure those names would be appropriate, since the consumer could definitely create a filled or partially-filled model through a closure, and there would not be anything non-default that seems like it would call for using "default". Is there something you're thinking of that I'm missing?